File: Source Code/.Net Demo/Source/ToggleButton.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace DotNetDemo
{
    public class ToggleButton : DemoButton
    {
        public ToggleButton(Rectangle rect, Color col)
            : base(rect, col)
        {
            isPressed = false;
        }

        public override void Depress()
        {
            isPressed = !isPressed;
            if (isPressed)
            {
                demo.selectDestroyer = true;
            }
            else
            {
                demo.selectDestroyer = false;
            }
        }
    }
}

File: Source Code/ManagedDemo/ManagedDemo/ManagedDemo/ToggleButton.cpp
#include "stdafx.h"
#include "ToggleButton.h"


ToggleButton::ToggleButton(DemoRectangle rect, Texture2D* tex, DemoColour col) : DemoButton(rect, tex, col)
{
	isPressed = false;
}


ToggleButton::~ToggleButton(void)
{
}

void ToggleButton::Depress()
{
	isPressed = !isPressed;
	if (isPressed)
		demo->selectDestroyer = true;
	else
	{
		demo->selectDestroyer = false;
	}
}

void ToggleButton::Unpress()
{
}
